admin_ui/backend/api/agents.py

```
@@ -195,19 +195,43 @@ def routing_methods():
+
 @router.post("/agents", status_code=201)
 def create_agent(body: AgentIn, request: Request):
+    data = body.model_dump()
+    if not _engine_ok(data.get("provider"), data.get("extra_json")):
+        raise HTTPException(422, "agent must have a provider or a pipeline")
+    data["provider"] = (data.get("provider") or "").strip()
     try:
-        return _store().create(**body.model_dump())
+        return _store().create(**data)
     except ValueError as e:
         raise HTTPException(422, str(e))
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)